packages/dds/merge-tree/src/client.ts

```diff
@@ -67,7 +67,15 @@ function elapsedMicroseconds(trace: Trace) {
     return trace.trace().duration * 1000;
 }
 
-export class Client {
+/**
+ * @internal
+ * Emitted before this client's merge-tree normalizes its segments on reconnect, potentially
+ * ordering them. Useful for DDS-like consumers built atop the merge-tree to compute any information
+ * they need for rebasing their ops on reconnection.
+ */
+export type IClientEvents = (event: "normalize", listener: () => void) => void;
```

Comment:
i'm not against Client being an event emitter, but it doesn't align with the existing callback pattern, and i prefer we are consistent. Also probably something to do in another pr.

Reply:
yeah... I hated that inconsistency while I was writing it. The slightly awkward thing is that you can only attach one callback, but we could always work around that by making interval collections' subscription check for an existing callback and call it, effectively making a linked list. It's a bit icky but I don't think it'll be an issue as there probably aren't a ton of interval collections on any given shared string.

Reply:
ahh. i didn't notice the multiple listeners, i only saw one in interval collection. i'd also be fine with openning an issue to move the other callbacks to events

Reply:
just re-checked and there is only one listener? maybe one got removed in refactoring? still open to changing later, as i do something similar wrangling is tests to get multiple listeners

Reply:
the thing is each interval collection on the SharedString needs to attach to the underlying merge-tree event. There can be multiple. Alternatively we could have SharedSegmentSequence subscribe and invoke an API on IntervalCollection, but that seems more gross imo (users actually get their hands on IntervalCollection whereas they generally don't get their hands on a Client)

Reply:
i'm fine with this. lets just open an issue to move client to the event emitter model. there are some test cases this will simplify as well

Reply:
Sounds good. Filed AB#2224 for this.